network/src/peer_manager/builder.rs

```diff
@@ -273,35 +273,22 @@ impl PeerManagerBuilder {
 
         let protos = transport_context.supported_protocols();
         let chain_id = transport_context.chain_id;
-        let network_id = self.network_context.network_id().clone();
-        let peer_id = self.network_context.peer_id();
-
-        let (key, maybe_trusted_peers, peer_id) = match transport_context.authentication_mode {
-            // validator-operated full node
-            AuthenticationMode::ServerOnly(key) if peer_id == PeerId::ZERO => {
-                let public_key = key.public_key();
-                let peer_id = PeerId::from_identity_public_key(public_key);
```

Comment:
aren't you missing this edge case now?

Reply:
the peer LGTM except for this part that seems a bit tricky to me. I don't think we really enforce that your peer id is correctly derived from your pubkey in network where you're not authenticated

Reply:
there is really no enforcement, if you look, we only do this if peer_id is zero, if someone tweaks their peer_id, there's not much we can do... the configs already make it impossible to accidentally have a peer_id zero. This is some legacy code when we were trying to figure out should we enforce this in networking or in the configs. I opt for configs since no one should be directly poking at this code.

Reply:
sounds good!